src/overlays/ovl_i4/fox_sz.c

```diff
@@ -1,55 +1,1853 @@
+    s32 pad[2];
+    f32 var_fa0;
+    f32 var_fv1;
+
+    switch (++actor->iwork[9]) {
+        case 600:
+            Radio_PlayMessage(gMsg_ID_16080, RCID_ROB64);
+            break;
+
+        case 1000:
+            Radio_PlayMessage(gMsg_ID_16085, RCID_ROB64);
+            break;
+
+        case 1200:
+            Radio_PlayMessage(gMsg_ID_16090, RCID_ROB64);
+            break;
+    }
+
+    var_fv1 = gBosses->obj.pos.x - actor->obj.pos.x;
+    var_fa0 = gBosses->obj.pos.z - actor->obj.pos.z;
+    __sinf((gGameFrameCount * M_DTOR));
+
+    if (0) {
+        ;
+    }
+    if (var_fv1) {
+        ;
+    }
+    if (var_fa0) {
+        ;
+    }
+
+    if (actor->unk_0E4 < 100) {
+        var_fv1 = __sinf(((f32) ((actor->index * 45) + gGameFrameCount)) * M_DTOR) * 5000.0f;
+        var_fa0 = __cosf(((f32) ((actor->index * 45) + (gGameFrameCount * 2))) * M_DTOR) * 5000.0f;
+    } else {
+        var_fa0 = 0.0f;
+        var_fv1 = 0.0f;
+    }
+
+    actor->fwork[4] = (gBosses->obj.pos.x + var_fv1) + 400.0f;
+    actor->fwork[5] = (gBosses->obj.pos.y + var_fa0) + 100.0f;
+    // temp_fv0 = fabsf(actor->fwork[4] - actor->obj.pos.x);
+    actor->fwork[6] = gBosses->obj.pos.z;
+    actor->fwork[3] = 1.4f;
+
+    if (((fabsf(actor->fwork[4] - actor->obj.pos.x) < 800.0f) &&
+         (fabsf(actor->fwork[5] - actor->obj.pos.y) < 800.0f)) &&
+        (fabsf(actor->fwork[6] - actor->obj.pos.z) < 800.0f)) {
+        func_i4_80199900(actor, 0);
+        D_80178480 = 0x19;
+        gBosses->dmgType = 100;
+        if ((gPlayer->state_1C8 == PLAYERSTATE_1C8_3) || (gPlayer->state_1C8 == PLAYERSTATE_1C8_5)) {
+            gPlayer->state_1C8 = PLAYERSTATE_1C8_7;
+            gPlayer->unk_1D0 = 0;
+            gActors->state = -0x7960;
+            return;
+        }
+    }
+
+    if (((fabsf(actor->fwork[6] - actor->obj.pos.z) < 2000.0f) &&
+         (((gPlayer->camEye.z < 0.0f) || (D_801615D0.y < 0.0f)) || (gPlayer->state_1C8 == PLAYERSTATE_1C8_10))) &&
+        (((gPlayer->state_1C8 == PLAYERSTATE_1C8_3) || (gPlayer->state_1C8 == PLAYERSTATE_1C8_10)) ||
+         (gPlayer->state_1C8 == PLAYERSTATE_1C8_5))) {
+        gPlayer->state_1C8 = PLAYERSTATE_1C8_7;
+        gPlayer->unk_1D0 = 100;
+        gActors->state = -0x7960;
+    }
+}
+#else
 #pragma GLOBAL_ASM("asm/us/nonmatchings/overlays/ovl_i4/fox_sz/func_i4_80199C60.s")
+#endif
+
+void func_i4_80199FCC(Actor* actor, s32 arg1) {
+    Actor_Initialize(actor);
+    actor->obj.status = 1;
+    actor->obj.id = OBJ_ACTOR_197;
+    actor->unk_0E4 = 200;
+
+    actor->obj.pos.x = D_i4_8019F4C0[arg1].x;
+    actor->obj.pos.y = D_i4_8019F4C0[arg1].y;
+    actor->obj.pos.z = D_i4_8019F4C0[arg1].z;
+
+    actor->state = 5;
+    actor->unk_0F4.y = 180.0f;
+
+    Object_SetInfo(&actor->info, actor->obj.id);
+
+    actor->health = 250;
+    actor->info.drawType = 2;
+    actor->info.hitbox = SEGMENTED_TO_VIRTUAL(D_6009230);
+    actor->fwork[1] = 25.0f;
+    actor->fwork[0x1D] = 2.0f;
```

Comment:
```suggestion
    actor->fwork[29] = 2.0f;
```
Work offsets in decimal.